CondFormats/HGCalObjects/interface/HGCalESProducerTools.h

```diff
@@ -0,0 +1,97 @@
+#ifndef CondFormats_HGCalObjects_HGCalESProducerTools_h
+#define CondFormats_HGCalObjects_HGCalESProducerTools_h
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <string>
+#include <sstream>    // for std::istringstream
+#include <fnmatch.h>  // to match glob patterns
+#include <nlohmann/json.hpp>
+using json = nlohmann::ordered_json;  // ordered_json preserves key insertion order
+
+namespace hgcal {
+
+  // @short search first match to a given typecode key in a JSON (following insertion order)
+  // allow glob patterns, e.g. 'ML-*', 'M[LH]-[A-Z]3*', etc.
+  inline std::string search_modkey(const std::string& module, const json& data, const std::string& name = "") {
+    if (!data.is_object() or data.empty()) {
+      edm::LogError("search_modkey") << "search_modkey: '" << name
+                                     << "' does not have the expected map/dict structure!";
+    }
+    for (auto it = data.begin(); it != data.end(); ++it) {
+      int match = fnmatch(it.key().c_str(), module.c_str(), FNM_EXTMATCH);
+      if (match != FNM_NOMATCH) {  // found matching key !
+        edm::LogInfo("search_modkey") << "search_modkey: Matched module='\e[1m" << module << "'\e[0m to modkey='\e[1m"
+                                      << it.key() << "\e[0m'";
```

Comment:
We (used to ?) have a rule in CMSSW against the use of bold, colours, etc. in the log messages.

One reason is that it will potentially lead to an escalation of more and more "visible" messages.

Other, more technical reasons, are that the ANSI escape sequences are not portable (not all terminal supports them) and will show poorly in the web-based log viewers (like the logs in the IBs and PRs).

If you _really_ need to have bold or coloured text, I'd suggest to at least see if the [`fmt`](https://github.com/fmtlib/fmt) library can offer a portable solution.

Reply:
It was useful for testing. I just removed it now. 😔